guide/index.adoc

```diff
@@ -0,0 +1,929 @@
+You can also contact the Scale team for assistance with adding JobTypes that contain secrets.
+
+=== Optimization
+
+==== Image Optimization
+Use of Anaconda and other large libraries associated with data science and machine learning jobs is common. 
+This makes sense for development on Windows or other environments that make compilation of complex scientific and math libraries challenging but should not be used in a Seed job. 
+As a rule of thumb, in order to avoid excessive strain on Docker registry and image cache resources, Docker images in Seed jobs should be kept under 500 MB. 
+Docker images should be based on minimal operating systems, such as {alpine-docker-url}[Alpine] or {busybox-docker-url}[Busybox], and layer optimization techniques should be applied. 
+Docker recommends a number of {docker-best-practices-url}[best practices] for writing Dockerfiles.
+
+==== Image Clean-up
+Another strategy for optimizing the size of the Docker image associated with the Seed job is to add commands to clean up after any package installations. 
+Consult the documentation for any package management system used by the operating system (e.g `apk` or `yum`) and/or language (e.g. `pip` or `mpm`), as the exact clean-up commands will vary depending on the environment and tools being used.
+
+=== Best Practices
+
+==== Workflow Decomposition
+Breaking down larger processing chains into separate jobs will yield benefits throughout the algorithm development lifecycle. 
+Problems can be discovered sooner and component algorithms can be built and tested independently, helping to minimize any single person or team being a bottleneck. 
+The greatest benefits can be realized when extracting general purpose, reusable jobs that can be applied to multiple data types, such as a job for generating a tile pyramid within a GeoTIFF. 
+This also helps ensure that commonly used jobs are both optimized and thoroughly proven by reuse. 
+Of course, there are always exceptions, and some algorithms (e.g. those that are I/O bound) may benefit from combining several processing steps into a single job.
+
+==== Log Everything
+Not having direct access to the file system of your job means the only method for feedback on what is happening inside the container is through console output. 
+Use standard output and/or standard error to indicate any progress or errors for your algorithm. 
+Some languages (such as Python) may require specifying that output should not be buffered until the process exits. 
+This will facilitate live viewing of output for long-running processes.
+
+==== Meaningful Errors
+Specific error conditions represented by unique process exit codes should be used to give clear feedback when a known error has occurred. 
+These can be used to distinguish between data errors where the data arrived but contained an error that could not be automatically addressed by the job and errors that occurred within the job execution itself. 
```

Comment:
>These can be used to distinguish between data errors (where the data arrived but contained an error that could not be automatically addressed by the job) and errors that occurred within the job execution itself.

Reply:
Updated in [df17722](https://github.com/ngageoint/seed-cli/pull/233/commits/df1772214e5dfcaf9c9bf7d76daa0a62fda86b0c)